proposals/NNNN-atf-improvment-proposal.md

```diff
@@ -0,0 +1,79 @@
+
+* Proposal: [SDL-NNNN](NNNN-atf-improvement-proposal.md)
+* Author: [Alexander Kutsan](https://github.com/smartdevicelink)
+* Status: **Awaiting review**
+* Impacted Platforms: [Core / ATF]
+
+## Introduction
+
+The proposal is to make ATF more flexible, salable and easy to use and integrate with popular CI systems and to analyze and full of all required information. 
+Also some functionality of ATF should be moved to scripts common files due to it's dependency from SDL version and features.
+
+## Motivation
+
+Current ATF requires and unfortunately allows changes during testing specific SDL scenarios. 
+Also it requires additional environment preparation and management during testing test sets
+and integrating ATF to popular CI systems.
+Need to make ATF more flexible and easy to use. And extend it scaling possibilities.
+A lot of common functionality should be moved from ATF to scripts common. 
+
+## Proposed solution
+
+There is 3 different layers of using ATF:
+
+1. Local executing and debugging concrete scenario even on local SDL build and script
+2. Local test set execution, Logs, reports, traffic analyses. Set set can be executed on remote SDL and on local SDL build
+3. CI execution of test set, collection Logs, reports, traffic logs for further analysis 
+
+ATF should be easy to use as for executing and debugging one concrete scenario on local workstation even with customly modified SDL.
+
+ATF architecture should be modified to support all this 3 layers of execution. Also ATF should support testing SDL on remote workstation/target. 
+
+Automatic creating connection and mobile session with SDL, and HMI - SDL connection should be removed from ATF.
+Many scripts do not need HMI or mobile connection, or app registration with specific parameters. This functionality should be part of scripts, but not ATF.
+
+ATF output artifacts should be improved.
+ATF should have provide test engineer ability to log custom messages
+Proposed output artifacts of ATF:
+ * Transport logs with time stamps and test case marks
+ * Collected via telnet SDL logs
+ * ATF junit (or other popular report format) report (in case if test set was executed) 
+ * ATF internal log with custom messages and expectations analyses
+ * Test script artifacts (files, used in test script)
+ 
+ATF should not fail script due to internal logic. In case if something unusual occurs (SDL is down or connection terminated) ATF should provide callback to script, and user should decide if script should be failed in such case. 
+
+ATF should provide interface for pre and post condition for scripts. Post condition should be executed even if test was failed. 
+
+ATF should deploy clear SDL environment for each script run.
+
+## Detailed design
+
+Split ATF to parts:
+ * API Facade
+ * script runner
+ * Test set runner
+ * SDL Watchdog
+ * SDL Deployer 
+
+All this parts of ATF should be partially standalone applications.
+ATF should manage with it's with of git modules system. 
```

Comment:
> ATF should manage with it's with of git modules system. 

 ATF should manage with it's with - With what? Rephrase the sentence please, it makes no sense to me.